torch/distributed/_sharded_tensor/api.py

```diff
@@ -336,110 +329,145 @@ def _init_rpc(self):
     def _init_from_local_shards(
         cls,
         local_shards: List[Shard],
-        sharded_tensor_metadata: ShardedTensorMetadata,
+        *overall_size,
         process_group=None,
         init_rrefs=False,
     ):
-        shards_metadata = sharded_tensor_metadata.shards_metadata
-        tensor_properties = sharded_tensor_metadata.tensor_properties
-
-        if len(shards_metadata) == 0:
-            raise ValueError("shards_metadata must not be empty!")
-
-        if tensor_properties.layout != torch.strided:
-            raise ValueError('Only torch.strided layout is currently supported')
-
-        sharded_tensor = cls.__new__(cls)
-
         # prepare initialization
+        sharded_tensor = cls.__new__(cls)
```

Comment:
nit: here we New ShardedTensor b/f the construct param is ready (eg., sharded tensor metadata). 

does it make sense to do validation first to collect all required params first
once validation and all_gather are done -- we create sharded tensor and init it right way and return?